### PR TITLE
Implement fizzbuzz in cairo

### LINKED ISSUE #216: [FEAT]: FizzBuzz Implementation in Cairo
# FizzBuzz Implementation in Cairo
## 📘 Description
Create an example that implements the classic "FizzBuzz" problem in Cairo. This is one of the most common coding exercises used in interviews to test basic programming skills. The task is to print numbers from 1 to n, but for multiples of 3 print "Fizz" instead, for multiples of 5 print "Buzz", and for multiples of both 3 and 5 print "FizzBuzz".

## ✅ Acceptance Criteria
- Create an `exercises/fizzbuzz` folder inside scripts folder and there create a scarb project as the other scripts.
- Implement a function that takes a number n and returns an array of strings representing the FizzBuzz sequence from 1 to n.
- The function should follow these rules:
  - For numbers divisible by 3, the array should contain "Fizz"
  - For numbers divisible by 5, the array should contain "Buzz"
  - For numbers divisible by both 3 and 5, the array should contain "FizzBuzz"
  - For all other numbers, the array should contain the string representation of the number
- Include comprehensive test cases that verify the implementation works for various input values.

### Example 1:
**Input**: n = 3  
**Output**: ["1", "2", "Fizz"]  
**Explanation**: 3 is divisible by 3, so we replace it with "Fizz".

### Example 2:
**Input**: n = 5  
**Output**: ["1", "2", "Fizz", "4", "Buzz"]  
**Explanation**: 5 is divisible by 5, so we replace it with "Buzz".

### Example 3:
**Input**: n = 15  
**Output**: ["1", "2", "Fizz", "4", "Buzz", "Fizz", "7", "8", "Fizz", "Buzz", "11", "Fizz", "13", "14", "FizzBuzz"]  
**Explanation**: 15 is divisible by both 3 and 5, so we replace it with "FizzBuzz".

## 🌎 References
- https://book.cairo-lang.org/
- https://en.wikipedia.org/wiki/Fizz_buzz

## 📜 Additional Notes
- ⚠ Read our guidelines before applying.
- This is a beginner-friendly exercise that helps practice basic control flow and string operations in Cairo.
- Consider how to handle string operations efficiently in Cairo.
- Think about how to optimize the solution to minimize gas costs.

### PR DESCRIPTION
**Pull request by GideonBature**
This PR implements the classic FizzBuzz problem in Cairo as requested in issue #216. The implementation follows the standard rules:
- For numbers divisible by 3, print "Fizz"
- For numbers divisible by 5, print "Buzz"
- For numbers divisible by both 3 and 5, print "FizzBuzz"
- Otherwise, print the number itself

## Implementation Details
- Created a new Scarb project in `examples/cairo/scripts/fizzbuzz/`
- Implemented the main `fizzbuzz` function that takes a number `n` and returns an array of strings from 1 to n
- Created a utility function `u32_to_felt252` to convert numbers to string representation
- Added comprehensive test cases covering multiple edge cases

## File Structure
- `.tool-versions`: Specifies Scarb version 2.11.2
- `Scarb.toml`: Project configuration
- `src/fizz_buzz.cairo`: Main FizzBuzz implementation
- `src/utils/number_converter.cairo`: Utility functions for number-to-string conversion
- `tests/test_fizzbuzz.cairo`: Test cases validating the implementation

## Testing
The implementation includes tests for various input values:
- Test with n=3: Validates handling of numbers 1-3, including the "Fizz" case
- Test with n=5: Validates handling of numbers 1-5, including both "Fizz" and "Buzz" cases
- Test with n=15: Comprehensive test covering all possible cases including "FizzBuzz"

All tests pass successfully, confirming the implementation meets the requirements specified in issue #216.

## 🖼️ Screenshots (if applicable)

Screenshot 2025-04-25 at 1 08 05 PM

## Related Issue
Closes #216

### CODE REVIEW

**Comment by GideonBature** [comment]
@coxmars This PR is ready for review and to be merged.

File: examples/cairo/scripts/fizzbuzz/src/utils/number_converter.cairo [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        let digit = *reversed_digits.at(index);
+        result = result * 256 + digit; // Shift left by 8 bits (1 byte) and add the next digit
+        i += 1;
+    }
+
+    result
+}
+
+/// Utility function to calculate base^exp
+fn pow(base: felt252, exp: u32) -> felt252 {
```
Comment (coxmars) [inline examples/cairo/scripts/fizzbuzz/src/utils/number_converter.cairo:44]: Where is the purpose of this pow function in this FizzBuzz impl?

Also, we have a simple math functions in case you need it: https://github.com/KaizeNodeLabs/starkiro/blob/4ed08c0d0d3083f770270bf83d03dd5b231e3c4a/examples/cairo/scripts/simple_math/src/simple_math.cairo#L23

**Review by coxmars — CHANGES_REQUESTED** [review]
@GideonBature Just adjust what I told you, if you are not using it just delete it or if you use it try to change that for our own functions from simple_math than creating a new one.

**Comment by GideonBature** [comment]
> @GideonBature Just adjust what I told you, if you are not using it just delete it or if you use it try to change that for our own functions from simple_math than creating a new one.

True, almost forgot that, was playing around with it while trying out other implementations, till I got a better way of using ASCII. Thank you for the review. Done as requested.

**Review by coxmars — APPROVED** [review]
LGTM 🫡
